src/App.js

```diff
@@ -2,28 +2,37 @@ import React, { useState, useEffect } from "react";
 import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 
-function useSemiPersistentState() {
-
-  const [todoList, setTodoList] = useState(() => {
-    const savedTodoList = localStorage.getItem("savedTodoList");
-    return savedTodoList ? JSON.parse(savedTodoList) : [];
+function App() {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const loadData = new Promise ((resolve, reject) => {
+      setTimeout(() => {
+        resolve({
+          data: {
+            todoList: JSON.parse(localStorage.getItem("savedTodoList"))
+          }
+        })
+      }, reject, 2000);
+    });
+    loadData.then((result) => {
+      setTodoList(result.data.todoList);
+      setIsLoading(false);
     });
-  
-    useEffect(() => {
-      localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-    }, [todoList]);
 
-    return [todoList, setTodoList]
-}
+  });
+  useEffect(() => {
+    if (!isLoading) {
+      localStorage.setItem("savedTodoList", JSON.stringify(todoList));
+    }
 
-function App() {
+  }, [todoList, isLoading]);
```

Comment:
Nice work adding `isLoading` to the dependency array to address the compilation error even though the instructions failed to mention it.